docs/concepts/rest_apis.rst

```diff
@@ -0,0 +1,32 @@
+edx-platform REST API Concepts
+##############################
+
+APIs in the edx-platform fall into one of two categories.  **Personal APIs**
+that only let you manipluate resources related to your user (the single user
+associated with the OAuth2 Application) or **Machine-to-machine APIs** that
+allow you to manipulate other users and system resources so long as the user
+associated with the OAuth2 application has the permissions to do so.
+
+The best way to interact with the APIs is to get a JWT Token associated with a
+user and then pass that to the server as a part of the request header.
+
+You can get a JWT one of two ways, one is to exchange the username and password
+for a user to get their JWT, and the other is to get a JWT associated with an
+OAuth2 Application (the application is associated with your user)that allow you
+to manipulate other users and system resources so long as the user associated
+with the OAuth2 application has the permissions to do so.
+
+The best way to interact with the APIs is to get a JWT Token associated with a
+user and then pass that to the server as a part of the request header.
+
+You can get a JWT one of two ways, one is to exchange the username and password
+for a user to get their JWT, and the other is to get a JWT associated with an
+OAuth2 Application (the application is associated with your user).
+
+JWTs by default expire every hour so when they expire you'll have to get a new
+one before you can call the API again.
```

Comment:
Do you mind to add a code sample to the how-to that shows how you detect that a JWT is expired, and how to refresh it?

Reply:
@pomegranited: Refreshing tokens is a different idea which I am _guessing_ Feanil was ignoring for the sake of this document. He is simply asking them to get a new one using the same original credentials, which I think is fine. I'm not sure if it will add confusion or not to mention that there is also a way to refresh using the refresh token, but that adds complexity, and would probably go in the other how to anyway.

Reply:
I ended up adding a reference section of useful code that includes using refresh tokens if you are working with a public client (you don't get refresh tokens with your JWT for `Confidential` clients).  Take a look at the new `auth_code_samples.rst`.